ferrocene/doc/internal-procedures/src/testing-other-targets.rst

```diff
@@ -0,0 +1,254 @@
+installed, as well as a few extras:
+
+.. code-block:: bash
+
+   sudo apt install qemu-user-static binfmt-support
+
+
+:target-with-triple:`aarch64-apple-darwin`
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Install Lima, if you don't have it:
+
+.. code-block:: bash
+
+    brew install lima
+
+You can create a guest with the following:
+
+.. code-block:: yaml
+
+    cat <<- EOF | limactl create --name=ferrocene
+        minimumLimaVersion: "1.0.0"
+        images:
+        - location: "https://cloud-images.ubuntu.com/releases/24.10/release-20241212/ubuntu-24.10-server-cloudimg-arm64.img"
+            arch: "aarch64"
+            digest: "sha256:fb39312ffd2b47b97eaef6ff197912eaa3e0a215eb3eecfbf2a24acd96ee1125"
+        - location: "https://cloud-images.ubuntu.com/releases/24.10/release/ubuntu-24.10-server-cloudimg-arm64.img"
+            arch: "aarch64"
+        mounts:
+        - location: "~"
+        - location: "/tmp/lima"
+        writable: true
+        ssh:
+        forwardAgent: true
+        cpus: 8
+        memory: "8GiB"
+        mountTypesUnsupported: ["9p"]
+    EOF
+
+Start the guest:
+
+.. code-block:: bash
+    
+    limactl start ferrocene
+
+
+Shell into the guest:
+
+.. code-block:: bash
+    
+    limactl shell ferrocene
+
+You can also point `Visual Studio Code's SSH extension <https://code.visualstudio.com/docs/remote/ssh>`_ at it
+using `these steps <https://github.com/lima-vm/lima/discussions/1890#discussioncomment-7221563>`_.
+
+Finally, ensure the guest is configured according to :doc:`internal-procedures:setup-local-env` as well as the :target-with-triple:`x86_64-unknown-linux-gnu` on this page.
+
+.. Warning::
+    
+    It is recommended to not share ``build/`` directories between multiple hosts, both for performance and correctness. To avoid this,
+    you should ``cd ~`` in the guest and clone a new copy of the Ferrocene repository into the dedicated guest storage.
+
+    Please ensure you always work from the guest-local repository.
+
+:target-with-triple:`x86_64-pc-windows-msvc`
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Setup WSL2, if you don't have it:
+
+.. code-block:: bash
+
+    wsl --install --distribution Ubuntu-24.04
+
+Ensure ```nestedVirtualization`` is set in the guest ``/etc/wsl.conf``, here is an example
+configuration:
+
+.. code-block::
+
+    [user]
+    default=ana
+
+    [boot]
+    systemd=true
+
+    [wsl2]
+    nestedVirtualization=true
+    
+If you changed your configuration, make sure to restart the environment with ``wsl --shutdown``.
+
+Shell into the guest:
+
+.. code-block:: bash
+    
+    wsl
+    
+You can also point `Visual Studio Code WSL extension <https://code.visualstudio.com/docs/remote/wsl-tutorial>`_ at it.
+
+Finally, ensure the guest is configured according to :doc:`internal-procedures:setup-local-env` as well as the :target-with-triple:`x86_64-unknown-linux-gnu` on this page.
+
+.. Warning::
```

Comment:
This warning comes back several times and concerns all hosts; maybe we should move it to the start.

Reply:
It's unfortunately so important that I want to make sure people bump into it even if they skip parts of the document. :(